api/app/controllers/spree/api/taxons_controller.rb

```diff
@@ -4,21 +4,10 @@ module Spree
   module Api
     class TaxonsController < Spree::Api::BaseController
       def index
-        if taxonomy
-          @taxons = taxonomy.root.children
-        elsif params[:ids]
-          @taxons = Spree::Taxon.accessible_by(current_ability, :read).where(id: params[:ids].split(','))
-        else
-          @taxons = Spree::Taxon.accessible_by(current_ability, :read).order(:taxonomy_id, :lft).ransack(params[:q]).result
-        end
-
-        @taxons = paginate(@taxons)
+        @taxons = paginate(taxons)
```

Comment:
It is a bit twisted for me that `@taxons` will contains the paginated version of `taxon` which sets an ivar named `@taxons` which contains the taxon scope. 

What about naming the memoizing method `taxons_scope` or `all_taxons` or something similar?